src/mysql-util/src/aws_rds.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use aws_config::{Region, SdkConfig};
+use aws_credential_types::provider::error::CredentialsError;
+use aws_credential_types::provider::ProvideCredentials;
+use aws_sigv4::http_request::{
+    self, SignableBody, SignableRequest, SignatureLocation, SigningError, SigningSettings,
+};
+use aws_sigv4::sign::v4::signing_params::BuildError;
+use aws_sigv4::sign::v4::SigningParams;
+use std::time::Duration;
+
+const AWS_TOKEN_EXPIRATION_SECONDS: u64 = 900;
+const AWS_SERVICE: &str = "rds-db";
+const DB_ACTION: &str = "connect";
+
+#[derive(Debug, thiserror::Error)]
+pub enum RdsTokenError {
+    /// AWS SdkConfig did not contain a credentials provider
```

Comment:
tiniest nit: We (at least in the Adapter part of the code base) generally try and make all comments formal sentences, so capitalizing all comments, and finishing them with periods. I first thought this was overkill but it does lead to a nice consistent style :)